### PR TITLE
[DGL-Go] Inference for Node Prediction Pipeline (full & ns)

### PR DESCRIPTION
**Pull request by mufeili**
## Description

I'll continue working on a few functionalities:

- Inference on a node subset based on node binary mask name specified in the configuration file
- Report metric values if node labels are available
- Dump raw logits in addition to the hard classification prediction
- Save the random split used in training for later inference

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [ ] Code is well-documented
- [x] To the best of my knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

**Review by jermainewang — APPROVED** [review]
Approved other than one comment.

File: dglgo/README.md [hunk 1]
```diff
@@ -63,14 +63,16 @@ generate a configurate file `cora_sage.yaml` which includes:
...
 
-Below shows the configuration file generated by the command above.
+The snippet below shows the configuration file generated by the command above.
 
 ```yaml
 version: 0.0.1
-pipeline_name: nodepred
+pipeline:
+  name: nodepred
+  mode: train
```
[the remark this answers is not in the document]
Comment (mufeili) [reply dglgo/README.md:75]: done

File: dglgo/README.md [hunk 2]
```diff
@@ -396,6 +399,5 @@ all the available pipelines.
 A: Currently not supported. We will enable this feature soon. Please stay tuned!
 
 **Q: After training a model on some dataset, how can I apply it to another one?**
-A: The `save_path` option in the generated configuration file allows you to specify where
-to save the model after training. You can then modify the script generated by `dgl export`
+A: The `save_path` option in the generated configuration file allows you to specify the directory to save the experiment results. After training, `{save_path}/run_{i}.pth` will be the checkpoint for the i-th run, consisting of the training configuration and trained model state dict. You can then modify the script generated by `dgl export`
```
[the remark this answers is not in the document]
Comment (mufeili) [reply dglgo/README.md]: done
